wandb/sdk/artifacts/storage_handlers/fsspec_file_handler.py

```diff
@@ -0,0 +1,146 @@
+"""Storage handler utilizing fsspec."""
+import os
+import time
+from typing import TYPE_CHECKING, Any, Optional, Sequence, Union
+from urllib.parse import ParseResult
+
+from wandb.errors.term import termlog
+from wandb.sdk.artifacts.artifact_file_cache import get_artifact_file_cache
+from wandb.sdk.artifacts.artifact_manifest_entry import ArtifactManifestEntry
+from wandb.sdk.artifacts.storage_handler import DEFAULT_MAX_OBJECTS, StorageHandler
+from wandb.sdk.lib import filesystem
+from wandb.sdk.lib.hashutil import B64MD5, md5_string
+from wandb.sdk.lib.paths import FilePathStr, StrPath, URIStr
+
+if TYPE_CHECKING:
+    import fsspec  # type: ignore
+
+    from wandb.sdk.artifacts.artifact import Artifact
+
+
+class FsspecFileHandler(StorageHandler):
+    """Handles a variety of different storage solutions."""
+
+    def __init__(self, scheme: Optional[str] = None) -> None:
+        """Track files or directories on a variety of filesystem.
+
+        For now, this handler supports oss (Alibaba Object Storage System)
+        A list of all available options can be found under the following link:
+        https://github.com/fsspec/filesystem_spec/blob/master/fsspec/registry.py
+        """
+        self._scheme = scheme or "oss"
+        self._cache = get_artifact_file_cache()
+
+    def can_handle(self, parsed_url: "ParseResult") -> bool:
+        return parsed_url.scheme in self._scheme
```

Comment:
`self._scheme` is a `str`, so this will be true e.g. for `os://some_file`.
```suggestion
        return parsed_url.scheme == self._scheme
```

Reply:
My previous intention (which I switched back and forth) was to do something like this 
```
return parsed_url.scheme in self._fsspec.available_protocols()
```
This would allow all currently supported storage solutions from fsspec. However, I only ran "integration" tests on oss and hf.
Since the fsspec_handler is checked last, a s3 or gcs URI would still be handled using the dedicated, existing s3_handler.py or gcs_handler.py